static/js/stream_list.js

```diff
@@ -335,7 +339,10 @@ class StreamSidebarRow {
 
     update_unread_count() {
         const count = unread.num_unread_for_stream(this.sub.stream_id);
-        update_count_in_dom(this.$list_item, count);
+        const stream_has_any_unread_mention_messages = unread.stream_has_any_unread_mentions(
+            this.sub.stream_id,
+        );
+        update_count_in_dom(this.$list_item, count, stream_has_any_unread_mention_messages);
```

Comment:
Completed the TODO added by you here; and handled the case properly of pinning/unpinning any stream.